checksource-gen/app/Main.hs

```diff
@@ -17,18 +20,22 @@ main :: IO ()
 main = do
   pj@PlanJson {..} <- findAndDecodePlanJson
         $ ProjectRelativeToDir "/home/runner/submission"
-  let rootUnit = (pjUnits Map.!) $ head $ Set.toList $ planJsonIdRoots pj
+  let rootUnit = (pjUnits Map.!)
+        . fromJust
+        . L.find (\(Plan.UnitId text) -> T.isPrefixOf (T.pack "submission") text)
+        . Set.toList
+        $ planJsonIdRoots pj
   let deps = Set.toList
         $ ciLibDeps $ uComps rootUnit Map.! CompNameExe "main"
   mgr <- TLS.newTlsManager
   pds <- mapM (askHackageForUnitPD' mgr pj)
     $ filter (\ Plan.Unit { .. } -> case uPId of
                  Plan.PkgId (Plan.PkgName pkgname) _
                    -> pkgname /= "ghc-boot-th")
-    $ map (pjUnits Map.!) deps
+    $ map (\dep -> fromJust (Map.lookup dep pjUnits)) deps
```

Comment:
Could you explain the rationale for this?

Reply:
It was for getting call stack on error, but not necessary after the fix in `rootUnit`. Shall I revert it?

Reply:
Hmm... please.

Reply:
I would revert it before 21:00, or feel free to take it over if you like quick change.

Reply:
Thanks for your swift work!